projects/packages/forms/src/dashboard/inbox/dataviews/actions.js

```diff
@@ -14,10 +14,13 @@ export const BULK_ACTIONS = {
 
 export const viewAction = {
 	id: 'view-response',
-	label: __( 'View response', 'jetpack-forms' ),
+	icon: <Icon icon={ seen } />,
 	isPrimary: true,
-	icon: <Icon icon={ commentContent } />,
-	modalHeader: __( 'Response', 'jetpack-forms' ),
+	label: __( 'View response', 'jetpack-forms' ),
+};
```

Comment:
I find this change quite confusing. We have an action with no callback and we leave the consumer later on to add it conditionally. 

IMO it would be clearer if we have a `viewAction` and a `selectAction`. The latter would be a hook (for example like [here](https://github.com/WordPress/gutenberg/blob/trunk/packages/edit-site/src/components/dataviews-actions/index.js#L17)) and pass what we need (selection, onChangeSelection) to that hook.

We could also have a single action `ViewOrSelect` as a hook and pass what we need and add conditionally the callback or the Modal.

What do you think?

Reply:
I think you're right; I'll have somet time next week to shape up the code to clearer state.